multiapps-controller-core/src/main/java/org/cloudfoundry/multiapps/controller/core/parser/StagingParametersParser.java

```diff
@@ -42,9 +47,39 @@ public Staging parse(List<Map<String, Object>> parametersList) {
                                .healthCheckHttpEndpoint(healthCheckHttpEndpoint)
                                .isSshEnabled(isSshEnabled)
                                .dockerInfo(dockerInfo)
+                               .lifecycleType(lifecycleType)
                                .build();
     }
 
+    private LifecycleType parseLifecycleType(List<Map<String, Object>> parametersList) {
+        String lifecycleValue = (String) PropertiesUtil.getPropertyValue(parametersList, SupportedParameters.LIFECYCLE, null);
+        if (lifecycleValue == null) {
+            return null;
+        }
+        try {
+            return LifecycleType.valueOf(lifecycleValue.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new SLException("Unsupported lifecycle value: " + lifecycleValue);
+        }
+    }
+
+    private void validateLifecycleType(LifecycleType lifecycleType, List<String> buildpacks, DockerInfo dockerInfo) {
+        if (lifecycleType == LifecycleType.CNB && (buildpacks == null || buildpacks.isEmpty())) {
+            throw new SLException("Buildpacks must be provided when lifecycle is set to 'cnb'.");
+        }
+        // Validate Docker-specific conditions
+        if (lifecycleType == LifecycleType.DOCKER) {
+            if (dockerInfo == null) {
+                throw new SLException("Docker information must be provided when lifecycle is set to 'docker'.");
+            }
+            if (buildpacks != null && !buildpacks.isEmpty()) {
+                throw new SLException("Buildpacks must not be provided when lifecycle is set to 'docker'.");
+            }
+        } else if (dockerInfo != null && lifecycleType != null) {
+            throw new SLException("Docker information must not be provided when lifecycle is set to " + lifecycleType + "'.");
+        }
```

Comment:
- Extract string messages into Messages class. 
- Replace throwing SLException with ContentException, this should avoid automatic retry from multiapps cli plugin when error is related with content of mta descriptor.
- Try to avoid nested if blocks by extracting logic into new methods.